pkg/distributor/distributor.go

```diff
@@ -1249,7 +1247,17 @@ func forReplicationSet[T any](ctx context.Context, d *Distributor, replicationSe
 		// Nothing to do.
 	}
 
-	return ring.DoUntilQuorum(ctx, replicationSet, d.queryQuorumConfig, wrappedF, cleanup)
+	return ring.DoUntilQuorum(ctx, replicationSet, d.queryQuorumConfig(ctx), wrappedF, cleanup)
+}
+
+func (d *Distributor) queryQuorumConfig(ctx context.Context) ring.DoUntilQuorumConfig {
+	logger := spanlogger.FromContext(ctx, d.log)
+
+	return ring.DoUntilQuorumConfig{
+		MinimizeRequests: d.cfg.MinimizeIngesterRequests,
+		HedgingDelay:     d.cfg.MinimiseIngesterRequestsHedgingDelay,
+		Logger:           logger,
```

Comment:
Have you considered conditionally enabling the logger here (e.g. via an experimental CLI flag)? My main concern is that looking at the changes you've done in https://github.com/grafana/dskit/pull/361, it looks it may log quite a lot.

Reply:
For query requests with zone-awareness enabled and three zones, we'll log four times, each at debug level:

* three messages, once when each zone's requests are starting
* one message when quorum is reached

If request minimisation is enabled, then there will only be three messages overall.

If zone-awareness is not enabled, then we'll log once for each instance we send a request to, but again at debug level.

Given the default log level is info, these messages won't be logged by default, and will only be attached to traces as events. Typical query traces in our production cells have thousands of spans, so I'm not too worried about a few extra events.

Happy to add a configuration option as a follow-up PR if you think it's needed though.